design/design-doc-composition-functions.md

```diff
@@ -175,32 +179,557 @@ metadata:
   name: example
 spec:
   compositeTypeRef:
-    apiVersion: database.example.org/v1alpha1
+    apiVersion: database.example.org/v1
     kind: XPostgreSQLInstance
+  # This Composition specifies functions instead of (P&T) resources.
+  # This is an array - you can specify a pipeline of Functions.
   functions:
-  - name: my-cool-function
-    type: Container
-    container:
-      image: xkpg.io/my-cool-function:0.1.0
+    # Each entry in the functions array has a name. This uniquely identifies
+    # the Function 'call' within the Composition.
+  - name: compose-xr-using-go-templates
+    # The functionRef tells the Composition which Function to use. This Function
+    # uses Go Templates (like Helm) to render composed resources.
+    functionRef:
+    - name: go-templates
```

Comment:
I believe this should not be an array.

Reply:
Thanks - I made and fixed this mistake in a bunch of places but missed this one. Will update.